backend/src/main/java/ro/utcn/sd/mdantonio/StackUnderflow/controller/PostController.java

```diff
@@ -0,0 +1,89 @@
+import ro.utcn.sd.mdantonio.StackUnderflow.dto.VoteDTO;
+import ro.utcn.sd.mdantonio.StackUnderflow.service.PostManagementService;
+import ro.utcn.sd.mdantonio.StackUnderflow.service.TagManagementService;
+import ro.utcn.sd.mdantonio.StackUnderflow.service.UserSessionManagementService;
+
+import java.util.List;
+
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+public class PostController {
+    private final PostManagementService postManagementService;
+    private final TagManagementService tagManagementService;
+    private final UserSessionManagementService userSessionManagementService;
+
+    @GetMapping("/posts")
+    public List<PostDTO> readAllQuestions() {
+        return postManagementService.listQuestions();
+    }
+
+
+    @PostMapping("/posts")
+    public PostDTO createPost(@RequestBody PostDTO dto) {
+        int currentUserId = userSessionManagementService.loadCurrentUser().getUserid();
+        int postType = PostDTO.convertStringPostTypeToIntPostType(dto.getType());
+        PostDTO response = postManagementService.addPost(postType, dto.getAuthor().getId(), dto.getParent(), dto.getTitle(), dto.getBody(), dto.getDate());
+        dto.getTags().forEach(tagId -> tagManagementService.attachTagToPost(currentUserId, tagId, response.getId()));
+        return postManagementService.findPostById(response.getId());
+    }
+
+
+    @PutMapping("/posts")
+    public PostDTO updatePost(@RequestBody PostDTO dto) {
+        int currentUserId = userSessionManagementService.loadCurrentUser().getUserid();
+        return postManagementService.updatePost(currentUserId, dto.getId(), dto.getBody());
+    }
+
+    @DeleteMapping("/posts")
+    public void deletePost(@RequestBody PostDTO dto) {
+        int currentUserId = userSessionManagementService.loadCurrentUser().getUserid();
+        postManagementService.removePost(currentUserId, dto.getId());
+    }
+
+    @GetMapping("/posts/title/{name}")
+    public List<PostDTO> readQuestionsByTitle(@PathVariable("name") String name) {
+        return postManagementService.listPostByTitle(name);
+    }
+
+    @GetMapping("/posts/tag/{tagTitle}")
+    public List<PostDTO> readQuestionsByTag(@PathVariable("tagTitle") String tagTitle) {
+        return postManagementService.listPostByTag(tagTitle);
+    }
+
+    @GetMapping("/posts/{postId}")
+    public PostDTO readPost(@PathVariable("postId") Integer postId) {
+        return postManagementService.findPostById(postId);
+    }
+
+    @GetMapping("/posts/editable")
+    public List<PostDTO> readAllAnswers() {
+        int currentUserId = userSessionManagementService.loadCurrentUser().getUserid();
+        return postManagementService.listEditablePosts(currentUserId);
+    }
+
+    @GetMapping("/posts/{postId}/answers")
+    public List<PostDTO> readAllAnswers(@PathVariable("postId") Integer postId) {
+        return postManagementService.listQuestionResponses(postId);
+    }
+
+    @PostMapping("/posts/vote")
```

Comment:
Here it would be nice to have the post ID inside the url (between the posts and the vote segments)